cmd/srrs/main_test.go

```diff
@@ -118,255 +120,449 @@ func TestMain(m *testing.M) {
+				})
+			}
+		})
+	})
+
+	t.Run("EndpointError", func(t *testing.T) {
+		a := require.New(t)
 
-				wg.Wait()
-			})
+		handshake := &api.Handshake{
+			Version: trcapi.DefaultVersion,
+			Token:   "wachtwoord1",
 		}
+		req, err := http.NewRequest(http.MethodGet, "http://"+defaultTCPAddress+"/"+path.Join("api", "v1"), nil)
+		a.NoError(err)
+		req.SetBasicAuth("", handshake.Token)
+
+		logger.Debug("Sending authentication request...")
+		resp, err := http.DefaultClient.Do(req)
+		a.NoError(err)
+		defer resp.Body.Close()
+		logger.Debug("Received " + resp.Status)
+		a.Equal(http.StatusNotFound, resp.StatusCode, "Received status: %s", resp.Status)
 	})
 
-	t.Run("SRRC->TRC/command", func(t *testing.T) {
-		for i := 0; i < messageCount; i++ {
-			t.Run(strconv.Itoa(i), func(t *testing.T) {
-				a = assert.New(t)
+	t.Run("TokenError", func(t *testing.T) {
+		a := require.New(t)
+
+		handshake := &api.Handshake{
+			Version: trcapi.DefaultVersion,
+			Token:   "theRightToken",
+		}
 
-				expected := &api.State{
-					Command: apitest.RandomCommand(),
+		falseToken := "theWrongToken"
+
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+
+		// go-routine and main function are swapped to allow early termination
+		go func() {
+			req, err := http.NewRequest(http.MethodGet, "http://"+defaultTCPAddress+"/"+webapi.AuthEndpoint, nil)
+			a.NoError(err)
+			req.SetBasicAuth("", falseToken)
+
+			logger.Debug("Sending authentication request...")
+			resp, err := http.DefaultClient.Do(req)
+			a.NoError(err)
+			defer resp.Body.Close()
+			if resp.StatusCode != http.StatusUnauthorized && resp.StatusCode != http.StatusForbidden {
+				if resp.StatusCode == http.StatusOK {
+					logger.Error("Authentication was successful while token was invalid: " + resp.Status)
+					t.Fail()
 				}
+				logger.Error("Authentication failed with wrong error code: " + resp.Status)
+			}
+			wg.Done()
+		}()
 
-				b, err := json.Marshal(expected.Command)
-				a.NoError(err)
+		logger.Debug("Waiting for connection on Unix socket...")
+		unixConn, err := netLst.Accept()
+		a.NoError(err)
 
-				req, err := http.NewRequest(http.MethodPost, "http://"+defaultTCPAddress+"/"+webapi.CommandEndpoint, bytes.NewReader(b))
-				a.NoError(err)
-				req.SetBasicAuth("", sessionKey)
+		logger.Debug("Establishing mock TRC connection, expecting authentication error")
+		trc := trctest.Connect(
+			unixConn, unixConn,
+			trctest.WithHandler(api.MessageTypeState, trctest.DefaultStateHandler),
+			trctest.WithHandler(api.MessageTypePing, trctest.DefaultPingHandler),
+			trctest.WithHandler(api.MessageTypeHandshake, trctest.DefaultHandshakeHandler),
+		)
 
-				errCh := make(chan error, 1)
-				go func() {
-					logger.Debug("Sending command to SRRS...")
-					resp, err := http.DefaultClient.Do(req)
-					if err != nil {
-						errCh <- err
-					}
-					defer resp.Body.Close()
+		logger.Debug("Sending handshake...")
+		trc.SendHandshake(handshake)
```

Comment:
Error return value of `trc.SendHandshake` is not checked

Reply:
but it is not important for the testcase